api/src/paths/project/{projectId}/survey/{surveyId}/template/upload.test.ts

```diff
@@ -0,0 +1,222 @@
+    params: {
+      projectId: 1,
+      surveyId: 2
+    },
+    body: {},
+    files: [
+      {
+        fieldname: 'media',
+        originalname: 'test.txt',
+        encoding: '7bit',
+        mimetype: 'text/plain',
+        size: 340
+      }
+    ]
+  } as any;
+
+  let actualStatus = 0;
+
+  const mockRes = {
+    status: (status: number) => {
+      actualStatus = status;
+
+      return {
+        send: () => {
+          // do nothing
+        }
+      };
+    }
+  } as any;
+
+  const mockNext = {} as any;
+
+  it('should throw a 400 error when files are missing', async () => {
+    sinon.stub(db, 'getDBConnection').returns(dbConnectionObj);
+
+    try {
+      const result = upload.uploadMedia();
+
+      await result({ ...mockReq, files: [] }, mockRes, mockNext);
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError.status).to.equal(400);
+      expect(actualError.message).to.equal('Missing upload data');
+    }
+  });
+
+  it('should throw a 400 error when more than 1 file uploaded', async () => {
+    sinon.stub(db, 'getDBConnection').returns(dbConnectionObj);
+
+    try {
+      const result = upload.uploadMedia();
+
+      await result({ ...mockReq, files: ['file1', 'file2'] }, mockRes, mockNext);
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError.status).to.equal(400);
+      expect(actualError.message).to.equal('Too many files uploaded, expected 1');
+    }
+  });
+
+  it('should throw a 400 error when projectId is missing', async () => {
+    sinon.stub(db, 'getDBConnection').returns(dbConnectionObj);
+
+    try {
+      const result = upload.uploadMedia();
+
+      await result({ ...mockReq, params: { ...mockReq.params, projectId: null } }, mockRes, mockNext);
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError.status).to.equal(400);
+      expect(actualError.message).to.equal('Missing required path param: projectId');
+    }
+  });
+
+  it('should throw a 400 error when surveyId is missing', async () => {
+    sinon.stub(db, 'getDBConnection').returns(dbConnectionObj);
+
+    try {
+      const result = upload.uploadMedia();
+
+      await result({ ...mockReq, params: { ...mockReq.params, surveyId: null } }, mockRes, mockNext);
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError.status).to.equal(400);
+      expect(actualError.message).to.equal('Missing required path param: surveyId');
+    }
+  });
+
+  it('should throw a 400 error when no sql statement returned', async () => {
+    sinon.stub(db, 'getDBConnection').returns(dbConnectionObj);
+
+    sinon.stub(survey_occurrence_queries, 'postSurveyOccurrenceSubmissionSQL').returns(null);
+
+    const result = upload.uploadMedia();
+
+    try {
+      await result(mockReq, mockRes, mockNext);
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError.status).to.equal(400);
+      expect(actualError.message).to.equal('Upload was not successful');
```

Comment:
why isnt this error equaling the `failed to generate insert sql statement` message?

Reply:
the error thrown gets caught by the try-catch and a new error gets thrown.  Something we could probably address across the board at some point